str/trtools/merge_str_runner.py

```diff
@@ -102,6 +128,39 @@ def main(
                         )['vcf.gz']
                     )
                 num_samples = num_samples + len(ids_2)
+        # similarly, if third and fourth file directories are specified - read in input file paths:
```

Comment:
Can we rewrite the inputs/logic here? Copy pasting the code arbitrarily could be replaced with something more elegant. I was thinking you could replace this `input1 sample1 input2 sample2...` logic with an argument that takes an arbitrary number of strings as arguments `sample1,input1 sample2,input2`, then iterate over those as 
```
for sample, input in arg.split(','):
    do the stuff
```

which is not elegant, but does mean that the same code will handle any number of inputs without rewriting the logic

Reply:
Also are the sample IDs expected to be different for each directory? I'm not super familiar with this process, but I assumed all the co-processed files would be shards within a consistent callset :/

Reply:
Thanks, yes sample IDs _**should**_ be different for each directory. I should probably add a warning if duplicate IDs are found